assets/js/components/StoreErrorNotice.js

```diff
@@ -16,22 +16,30 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
 /**
  * Internal dependencies
  */
 import Data from 'googlesitekit-data';
-import { STORE_NAME } from '../../datastore/constants';
-import { isPermissionScopeError } from '../../../../googlesitekit/datastore/user/utils/is-permission-scope-error';
-import ErrorText from '../../../../components/error-text';
+import { isPermissionScopeError } from '../googlesitekit/datastore/user/utils/is-permission-scope-error';
+import ErrorText from '../components/error-text';
 const { useSelect } = Data;
 
-export default function ErrorNotice() {
-	const error = useSelect( ( select ) => select( STORE_NAME ).getError() );
+export default function ErrorNotice( { storeName, shouldDisplayError = () => true } ) {
+	const error = useSelect( ( select ) => select( storeName ).getError() );
 
 	// Do not display if no error, or if the error is for missing scopes.
-	if ( ! error || isPermissionScopeError( error ) ) {
+	if ( ! error || isPermissionScopeError( error ) || ! shouldDisplayError( error ) ) {
 		return null;
 	}
 
 	return <ErrorText message={ error.message } reconnectURL={ error.data?.reconnectURL } />;
 }
+
+ErrorNotice.propTypes = {
+	storeName: PropTypes.string.isRequired,
```

Comment:
Please add a prop type definition for the (optional) `shouldDisplayError` function as well.